internal/plugin/platform.go

```diff
@@ -428,17 +429,37 @@ func (s *platformServer) Deploy(
 	internal := s.internal()
 	defer internal.Cleanup.Close()
 
+	// NOTE(izaak): Argmapper has two different ways to get a DeclaredResourcesResp:
+	// the value we provide below, or converting the pb.DeclaredResources supplied by core
+	// with a mapper function. In practice, it uses this one (which is important), but I don't
+	// know if we can rely on that behavior. The commented-out snippet below removes the other arg.
+	declaredResourcesResp := &component.DeclaredResourcesResp{}
+
+	// Remove the DeclaredResources arg from the list - we don't want to use that one.
+	//var realArgs []*proto.FuncSpec_Value
```

Comment:
What's the potential use case of `realArgs` here, if it wasn't commented out?

Reply:
Ah good catch - the idea here is to remove the DeclaredResourcesResp that came in over the wire, and swap in our own. If we don't remove the other one, then argmapper has two DeclaredResourcesResp components to choose from to satisfy the plugin's DeployFunc. 

In practice, argmapper always seems to choose the one we inject here, without removing the old one. I'm not sure if that's behavior we should be depending on though.

Reply:
Is the duplicated message because it exists on the Waypoint server proto generated by a plugin and in the sdk here? Why are there two?

Reply:
This is no longer necessary!